hadoop-ozone/cli-admin/src/main/java/org/apache/hadoop/hdds/scm/cli/datanode/VolumeSubCommand.java

```diff
@@ -0,0 +1,222 @@
+ */
+@Command(
+    name = "volumes",
+    description = "Display the list of volumes on the DataNode.",
+    mixinStandardHelpOptions = true,
+    versionProvider = HddsVersionProvider.class)
+public class VolumeSubCommand extends ScmSubcommand {
+
+  // Display it in JSON format.
+  @Option(names = { "--json" },
+       defaultValue = "false",
+       description = "Format output as JSON.")
+  private boolean json;
+
+  // Display it in TABLE format.
+  @Option(names = { "--table" },
+       defaultValue = "false",
+       description = "Format output as Table.")
+  private boolean table;
+
+  // PageSize refers to the number of items displayed per page
+  // in a paginated view.
+  @Option(names = { "--pageSize" },
+       defaultValue = "20",
+       description = "The number of volume information items displayed per page.")
+  private int pageSize;
+
+  // The current page.
+  @Option(names = { "--currentPage" },
+       defaultValue = "1",
+       description = "The current page.")
+  private int currentPage;
+
+  /**
+   * We have designed a new option called 'show',
+   * which includes two selectable configurations:
+   * 'failed' is used to display failed disks,
+   * and 'normal' is used to display normal disks.
+   */
+  @Option(names = { "--displayMode" },
+      defaultValue = "all",
+      description = "failed is used to display failed disks, " +
+      "normal is used to display normal disks.")
+  private String displayMode;
+
+  // The UUID identifier of the DataNode.
+  @Option(names = { "--uuid" },
+      defaultValue = "",
+      description = "failed is used to display failed disks, " +
+      "normal is used to display normal disks.")
+  private String uuid;
+
+  // The HostName identifier of the DataNode.
+  @Option(names = { "--hostName" },
+      defaultValue = "",
+      description = "failed is used to display failed disks, " +
+      "normal is used to display normal disks.")
+  private String hostName;
+
+  private SimpleDateFormat sdf = new SimpleDateFormat(
+      "EEE MMM dd HH:mm:ss Z yyyy", Locale.ENGLISH);
+
+  private static final String DATANODE_VOLUME_FAILURES_TITLE = "Datanode Volume";
+
+  private static final List<String>  DATANODE_VOLUME_FAILURES_HEADER = Arrays.asList(
+      "UUID", "Host Name", "Volume Name", "Volume Status", "Capacity / Capacity Lost",
+      "Failure Time");
+
+  private final String[] validModes = {"all", "normal", "failed"};
+
+  @Override
+  public void execute(ScmClient client) throws IOException {
+
+    validateDisplayMode(displayMode);
+
+    // Retrieve the volume data based on the conditions.
+    GetVolumeInfosResponseProto response =
+        client.getVolumeInfos(displayMode, uuid, hostName, pageSize, currentPage);
+
+    // Print the relevant information if the return value is empty.
+    if (response == null || CollectionUtils.isEmpty(response.getVolumeInfosList())) {
+      System.out.println("No volume data was retrieved.");
```

Comment:
I'm not sure this informational message is needed, but if you think so, at least move it after handling JSON format, which should output valid JSON even for empty results.